plugin-base/src/main/java/io/everytrade/server/plugin/impl/everytrade/parser/exchange/AnycoinExchangeSpecificParserV1.java

```diff
@@ -0,0 +1,269 @@
+        buyBean.setTransactionType(TransactionType.BUY);
+        buyBean.setType("BUY");
+
+        result.add(buyBean);
+        result.add(newBean);
+    }
+
+    private void prepareUnstakeBeansWithCurrency(AnycoinBeanV1 beanV1, AnycoinBeanV1 newBean, List<AnycoinBeanV1> result) {
+        currenciesToBeIgnored.removeLast();
+        newBean.setOperationType(beanV1.getOperationType());
+        newBean.setDate(beanV1.getDate().toString());
+        newBean.setMarketBase(beanV1.getCoin());
+        newBean.setBaseAmount(beanV1.getAmount().abs());
+        newBean.setTransactionType(TransactionType.UNSTAKE);
+        newBean.setType(beanV1.getOperationType().code);
+
+        result.add(newBean);
+    }
+
+    private void prepareStakeRewardBeans(AnycoinBeanV1 beanV1, AnycoinBeanV1 newBean, List<AnycoinBeanV1> result) {
+        newBean.setDate(beanV1.getDate().toString());
+        newBean.setMarketBase(beanV1.getCoin());
+        newBean.setBaseAmount(beanV1.getAmount());
+        newBean.setTransactionType(TransactionType.STAKING_REWARD);
+        newBean.setType(beanV1.getOperationType().code);
+
+        AnycoinBeanV1 stakeBean = new AnycoinBeanV1();
+        stakeBean.setDate(beanV1.getDate().plusSeconds(1).toString());
+        stakeBean.setMarketBase(beanV1.getCoin());
+        stakeBean.setBaseAmount(beanV1.getAmount());
+        stakeBean.setTransactionType(TransactionType.STAKE);
+        stakeBean.setType("STAKE");
+
+        result.add(stakeBean);
+        result.add(newBean);
+    }
+
+    private void prepareDepositBeans(AnycoinBeanV1 beanV1, AnycoinBeanV1 newBean, List<AnycoinBeanV1> result) {
+        newBean.setOperationType(beanV1.getOperationType());
+        newBean.setDate(beanV1.getDate().toString());
+        newBean.setMarketBase(beanV1.getCoin());
+        newBean.setBaseAmount(beanV1.getAmount());
+        newBean.setMarketQuote(beanV1.getCoin());
+        newBean.setQuoteAmount(beanV1.getAmount());
+        newBean.setTransactionType(TransactionType.DEPOSIT);
+        newBean.setType(beanV1.getOperationType().code);
+
+        result.add(newBean);
+    }
+
+    private void prepareWithdrawalBeans(AnycoinBeanV1 beanV1, AnycoinBeanV1 newBean, List<AnycoinBeanV1> result) {
+        newBean.setOperationType(beanV1.getOperationType());
+        newBean.setDate(beanV1.getDate().toString());
+        newBean.setMarketBase(beanV1.getCoin());
+        newBean.setBaseAmount(beanV1.getAmount());
+        newBean.setMarketQuote(beanV1.getCoin());
+        newBean.setQuoteAmount(beanV1.getAmount());
+        newBean.setTransactionType(TransactionType.WITHDRAWAL);
+        newBean.setType(beanV1.getOperationType().code);
+
+        result.add(newBean);
+    }
+
+
+    @Override
+    public Map<String, List<AnycoinBeanV1>> createGroupsFromRows(List<AnycoinBeanV1> rows) {
+        for (AnycoinBeanV1 row : rows) {
+            if (row.getOrderId() == null) {
+                row.setOrderId("0");
+            }
+        }
```

Comment:
Do not forget validate groups ( set unsupported row and message fields )
Fill the List of Ids in the Bean

Reply:
fixed